### PR TITLE
FTE Enhancement  - fte_total_census, fte_total_current, fte_total_end_of_term

### PR DESCRIPTION
**Pull request by andreabringhurst**
When reporting FTE it is often reported in 3 ways, undergraduate fte, graduate fte and total fte. Writing 1 query for all 3 of these requests makes the most sense and means that there are less queries to manage in utDataStoR.

I added 3 new queries to fte_total_census, fte_total_current, and fte_total_end_of_term.

Adding a query for total fte that is for current, one for census and one for eot will mirror what exists with headcount in utDataStoR. There is a headcount_current, headcount_census and headcount_eot.

- Add fte_total_census to utDataStoR to replace
fte_gr_cred_hour_census and fte_ug_cred_hour_census
have the new query report both ug, gr, and total fte by term and do it in only one query instead of two.

- Add fte_total_eot to utDataStoR to replace
fte_ug_cred_hour_eot
have the new query report both ug, gr, and total fte by term and do it in only one query so that we don't need to create additional queries for gr and total.

- Add fte_total_current to utDataStoR to replace 
fte_off_edify
have the new query report both ug, gr, and total fte by term, have it use snapshot data and do not hard code terms. Correct the problems with fte_off_edify,  it is pulling from the student level and not the course level like fte_total_current will.

This pull request will take care of issues #15 , #16 , and #17

### CODE REVIEW

**Comment by MattNickodemus** [comment]
@cdemkegit Will you add the SQL for off the daily snap shot to this? @andreabr-dixie Will you coordinate this with Craig? We can probably do this is a separate pull-request.

**Review by CSPierce01 — APPROVED** [review]
I agree with the changes and the new sql runs quickly.

**Review by JoyBaule — APPROVED** [review]
I have reviewed the changes for FTE Enhancement and run each query in Datagrip to test the output. The output aligns with current and historical Census and End of Term summary files returned by USHE.

File: inst/sql/fte/fte_total_census.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+         (SELECT a.term_id,
+                 ROUND(SUM(a.attempted_credits) / 15, 2) as census_undergrad_fte
+          FROM export.student_section_version a
+          WHERE a.is_enrolled IS TRUE
+            AND a.version_desc = 'Census'
+            AND a.course_level_id = 'UG'
+          GROUP BY a.term_id)
+
+SELECT b.term_desc,
+       d.census_undergrad_fte,
```
[the remark this answers is not in the document]
Comment (MattNickodemus) [reply inst/sql/fte/fte_total_census.sql]: This is done.

File: inst/sql/fte/fte_total_census.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
+FTE Graduate, Undergraduate, and Total - Census
```
[the remark this answers is not in the document]
Comment (MattNickodemus) [reply inst/sql/fte/fte_total_census.sql]: This is done.

File: inst/sql/fte/fte_total_current.sql
[the remark this answers is not in the document]
Comment (MattNickodemus) [reply inst/sql/fte/fte_total_current.sql:1]: This is done.

**Comment by MattNickodemus** [comment]
@cdemkegit @andreabr-dixie I don't see the sql that we use to pull FTE off the daily snapshot.

**Review by MattNickodemus — APPROVED** [review]

**Comment by andreabr-dixie** [comment]
> @cdemkegit @andreabr-dixie I don't see the sql that we use to pull FTE off the daily snapshot.

@MattNickodemus It has it's own issue (#23 ) and is in the "no status" section of the project. It is not covered in this pull request.

**Comment by andreabr-dixie** [comment]
@MattNickodemus based on your comments, it looks like we can move forward with this pull request, correct? 
I can see that this pull request is to merge to main. I would like to close this merge request and do a new pull request to the dev_branch, and reference this pull request. I would work with @cdemkegit to do this, so it doesn't interfere with his current work in merging the vignettes to the dev_branch. But once both this pull request and the vignette pull requests are on dev_branch, then he can move both of the changes to the main branch.

**Comment by cdemkegit** [comment]
I think we are going to table this pull request to milestone: [2024 Q4](https://github.com/ut-effectiveness/utDataStoR/pulls?q=is%3Aopen+is%3Apr+milestone%3A%222024+Q4%22)

**Review by cdemkegit — APPROVED** [review]

**Review by mkhassan-isu — APPROVED** [review]

**Comment by MattNickodemus** [comment]
@andreabringhurst @cdemkegit Where are we on this?

**Comment by andreabringhurst** [comment]
@MattNickodemus @cdemkegit Craig and I had planned to milestone this without these queries because it will require changes to the make_fte_function and the vignettes (which haven't been done yet).

**Comment by andreabringhurst** [comment]
We have pushed the milestone and this pull request will now be in the milestone. These are ready to go and have been approved at code review.

**Comment by andreabringhurst** [comment]
I added two more sql's to this pull request. I added fte_ipeds_12_month.sql and fte_ushe_annualized.sql.

**Comment by MattNickodemus** [comment]
@andreabringhurst It likes like we're still adding queries on this. Can you give me an update on what the status of this pull request is?

**Review by MattNickodemus — APPROVED** [review]
I approved this in code review 20241121

**Comment by andreabringhurst** [comment]
@cdemkegit @JoyBaule @CSPierce01 @mkhassan-isu @MattNickodemus  Please review these again and approve the pull request.

**Review by CSPierce01 — APPROVED** [review]
I have reviewed and agree with the changes.

**Review by JoyBaule — APPROVED** [review]
Code Review 11/21/2024 to acknowledge and approve the addition of new SQL written for IPEDS 12 Month FTE and USHE Annualized FTE.

**Review by cdemkegit — APPROVED** [review]
These are the files we reviewed in code review yesterday Nov 21. I haven't ran these independently but the code looks sound.

**Comment by andreabringhurst** [comment]
Everyone reviewed the pull request after discussing it a code review on 11/21.

**Pull request merged**